Helper/AvailablePaymentMethodsHelper.php

```diff
@@ -51,32 +50,42 @@ class AvailablePaymentMethodsHelper
      */
     private SerializerInterface $serializer;
 
+    /**
+     * @var Configuration
+     */
+    private Configuration $configuration;
+
     /**
      * AvailablePaymentMethodsHelper constructor.
      *
      * @param AvailablePaymentMethods $availablePaymentMethod
      * @param CacheInterface $cache
      * @param SerializerInterface $serializer
      * @param StoreManagerInterface $storeManager
+     * @param Configuration $configuration
      */
     public function __construct(
         AvailablePaymentMethods $availablePaymentMethod,
         CacheInterface $cache,
         SerializerInterface $serializer,
-        StoreManagerInterface $storeManager
+        StoreManagerInterface $storeManager,
+        Configuration $configuration
     ) {
         $this->availablePaymentMethod = $availablePaymentMethod;
         $this->cache = $cache;
         $this->storeManager = $storeManager;
         $this->serializer = $serializer;
+        $this->configuration = $configuration;
     }
 
     /**
      * @return bool
      */
     public function canInitialize(): bool
     {
-        return class_exists('Mobile_Detect');
+        return class_exists('Mobile_Detect') &&
+            $this->configuration->getApiKey() &&
```

Comment:
Can we implement strict type checking? Seems like a code smell.

Reply:
did you mean to wrap in isset() both of the new checks?